### PR TITLE
Don't error under wine if VirtualAlloc fails

### PR DESCRIPTION
**Pull request by Keno**
Wine has a bug that makes VirtualQuery return the wrong answer in
some situations. Technical details can be found in my post to the
Wine mailing list at [1].

[1] https://www.winehq.org/pipermail/wine-devel/2016-March/112188.html

### CODE REVIEW

File: ui/repl.c [hunk 1]
```diff
@@ -595,8 +595,21 @@ int main(int argc, char *argv[])
 {
     uv_setup_args(argc, argv); // no-op on Windows
 #else
+
+#if defined(_P64) && defined(JL_DEBUG_BUILD)
+static int is_running_under_wine()
+{
+  static const char * (CDECL *pwine_get_version)(void);
```
Comment (tkelman) [inline ui/repl.c]: indent

File: src/jitlayers.cpp [hunk 2]
```diff
@@ -187,7 +187,7 @@ void NotifyDebugger(jit_code_entry *JITCodeEntry)
     }
     __jit_debug_descriptor.first_entry = JITCodeEntry;
     __jit_debug_descriptor.relevant_entry = JITCodeEntry;
-    __jit_debug_register_code();
+    //__jit_debug_register_code();
```
Comment (yuyichao) [inline src/jitlayers.cpp]: ?

Comment (Keno) [inline src/jitlayers.cpp]: Oops, sorry, I made sure to leave that out originally, but when I fixed the indent, it snuck back. Will fix.

File: ui/repl.c [hunk 3]
```diff
@@ -595,8 +595,21 @@ int main(int argc, char *argv[])
 {
     uv_setup_args(argc, argv); // no-op on Windows
 #else
+
+#if defined(_P64) && defined(JL_DEBUG_BUILD)
```
Comment (tkelman) [inline ui/repl.c]: other than the current only place you're calling it, is anything about this specific to 64 bit or debug build? fine to leave it this way for now, just curious. we might need this elsewherr later, at which point we can change or remove the ifdef.

Comment (Keno) [inline ui/repl.c]: No, this detects wine generally, but since it's not used anywhere else, I want to avoid the warning.

**Comment by nalimilan** [comment]
Maybe leave a comment with an explanation of the bug and a link to the thread?

**Comment by Keno** [comment]
The failure on this branch is interesting. I didn't think it was related, so I went ahead and merged, but I'm not sure we've seen this one before:
```
	From worker 15:	  Test threw an exception of type DimensionMismatch
	From worker 15:	  Expression: norm(idct(dct(a)) - a) < 1.0e-8
	From worker 15:	  DimensionMismatch("tried to assign 2-element array to 2x2 destination")
	From worker 15:	   in throw_setindex_mismatch(::Array{Int64,1}, ::Tuple{Int64,Int64}) at .\operators.jl:342
	From worker 15:	   [inlined code] from .\operators.jl:394
	From worker 15:	   in _unsafe_batchsetindex!(::Array{Int64,2}, ::Array{Int64,1}, ::UnitRange{Int64}, ::UnitRange{Int64}) at .\multidimensional.jl:331
	From worker 15:	   [inlined code] from .\multidimensional.jl:302
	From worker 15:	   in Base.DFT.FFTW.r2rFFTWPlan{Complex{Float64},ANY,false,2}(::Array{Complex{Float64},2}, ::Base.DFT.FFTW.FakeArray{Complex{Float64},2}, ::UnitRange{Int64}, ::Int64, ::UInt32, ::Float64) at .\fft\FFTW.jl:547
	From worker 15:	   [inlined code] from .\boot.jl:331
	From worker 15:	   in #plan_dct#29(::Array{Any,1}, ::Any, ::Array{Complex{Float64},2}, ::UnitRange{Int64}) at .\fft\dct.jl:36
	From worker 15:	   [inlined code] from .\boot.jl:331
	From worker 15:	   in #plan_dct#41(::Array{Any,1}, ::Any, ::Array{Complex{Float64},2}) at .\fft\dct.jl:89
	From worker 15:	   in dct(::Array{Complex{Float64},2}) at .\fft\dct.jl:87
	From worker 15:	   [inlined code] from C:\projects\julia\test\dsp.jl:46
	From worker 15:	   in anonymous at .\<no file>:4294967295
	From worker 15:	   [inlined code] from .\essentials.jl:79
	From worker 15:	   in include_string(::ASCIIString, ::ASCIIString) at .\loading.jl:371
	From worker 15:	   in include_from_node1(::ASCIIString) at .\loading.jl:420
	From worker 15:	   [inlined code] from .\util.jl:179
	From worker 15:	   in runtests(::ASCIIString) at C:\projects\julia\test\testdefs.jl:7
	From worker 15:	   in (::Base.Serializer.__deserialized_types__.##7638)(::ASCIIString) at C:\projects\julia\test\runtests.jl:36
	From worker 15:	   in run_work_thunk(::Base.##258#260{Base.CallMsg{:call_fetch}}, ::Bool) at .\multi.jl:744
	From worker 15:	   [inlined code] from .\multi.jl:1040
	From worker 15:	   in (::Base.##257#259{Base.CallMsg{:call_fetch},TCPSocket})() at .\task.jl:59
	From worker 16:	     * examples              in  21.54 seconds, maxrss  435.05 MB
WARNING: Forcibly interrupting busy workers 
```

**Comment by tkelman** [comment]
It's mentioned under the subarray test failure issue. Possibly similar cause.

**Pull request merged**